onnxruntime/core/providers/cuda/tunable/cuda_tuning_context.cc

```diff
@@ -63,6 +63,20 @@ bool CudaTuningContext::IsTunableOpEnabled() const {
   return info_->enabled;
 }
 
+void CudaTuningContext::EnableTuning() {
```

Comment:
What happens if someone calls `EnableTuning()` but with tunable ops not enabled? The logs would indicate that tuning is enabled, but in reality they wouldn't (because tunable ops are disabled). This would be rather confusing. 

It might be cleaner to just have `EnableTunableOpAndTuning()` instead. Or enforce the semantics that enabling tuning automatically enabled tunable ops as well.

Reply:
As warned, it is an nop in this case. Because the tuning itself happens only on the first op usage. Not using it will not trigger the tuning by design.